project1/server/file_handler/my_file_handler.cpp

```diff
@@ -0,0 +1,98 @@
+//
+// Created by jake on 1/25/21.
+//
+
+#include "my_file_handler.h"
+#include <bits/stdc++.h>
+
+namespace server::file_handler{
+  bool file_handler::MyFileHandler::Delete(const std::string &filename) {
+
+    //convert filename to char*
+    int n = filename.length();
+    char fileName[n+1];
+    strcpy(fileName, filename.c_str());
+
+    //delete file
+    return remove(fileName);
+  }//Delete
+  bool file_handler::MyFileHandler::Put(const std::string &filename, const std::vector<uint8_t> &contents) {
+
+    //convert contents to string format
+    std::string str_contents(contents.begin(), contents.end());
+
+    //open file
+    try {this->os.open(filename);}
+    catch (std::exception &e) {
+      this->os.close();
+      return false;
+    }
+
+    //write contents to file
+    try{this->os << str_contents;}
+    catch (std::exception &e){
+      this->os.close();
+      return false;
+    }
+
+    //close file
+    this->os.close();
+    return true;
+  }//Put
+  bool file_handler::MyFileHandler::MakeDir(const std::string &name) {
+    //convert directory name to char*
+    int n = name.length();
+    char dir[n+1];
+    strcpy(dir, name.c_str());
+
+    if (mkdir(dir,0777)==-1) return false;
+    else return true;
+  }//MakeDir
+  bool file_handler::MyFileHandler::ChangeDir(const std::string &sub_folder) {
+
+    //convert subdirectory name to char*
+    int n = sub_folder.length();
+    char dir[n+1];
+    strcpy(dir, sub_folder.c_str());
+
+    if(chdir(dir)==-1) return false;
+    else return true;
+  }//ChangeDir
+  std::vector<uint8_t> file_handler::MyFileHandler::Get(const std::string &filename) const {
+    std::ifstream stream;
+
+    stream.open(filename);
```

Comment:
Yes, actually because this Get() is a const member function, I am not able to reference any member variables. It might be worth it to remove the const from Get()

Reply:
Right, I thought I you brought this up on Discord and I said that the `const` was an uninformed assumption on my part, and it was fine to remove it if necessary. In this case, I'd prefer either using local `fstream`s everywhere, or removing the `const` and using member `fstream`s everywhere. In fact, I think I slightly prefer the former, as I don't really see that having the member variables buys you anything in this case.

Reply:
Yeah, I like removing the fstream member variable altogether. Looks like it was only used in one other function anyways.